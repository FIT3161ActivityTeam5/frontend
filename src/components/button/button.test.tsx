import React from 'react';
import renderer from 'react-test-renderer';

import Button from './button';

describe('<Button />', () => {
  // Snapshot test.
  it('renders correctly', () => {
    const tree = renderer.create(<Button title="Confirm" onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
