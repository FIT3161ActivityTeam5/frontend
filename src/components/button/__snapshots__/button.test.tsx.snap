// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> applies custom styles correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(220, 38, 38, 1)",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_700Bold",
        },
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  >
    Confirm
  </Text>
</View>
`;

exports[`<Button /> applies custom text styles correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(124, 58, 237, 1)",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_700Bold",
        },
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  >
    Confirm
  </Text>
</View>
`;

exports[`<Button /> applies the loading indicator correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(124, 58, 237, 1)",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="rgba(249, 250, 251, 1)"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(124, 58, 237, 1)",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_700Bold",
        },
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  >
    Confirm
  </Text>
</View>
`;
